system_metrics_collector/CMakeLists.txt

```diff
@@ -41,11 +42,13 @@ add_library(system_metrics_collector SHARED
   src/moving_average_statistics/moving_average.cpp
   src/moving_average_statistics/types.hpp
 )
-
-ament_target_dependencies(system_metrics_collector rclcpp rcpputils)
+target_include_directories(system_metrics_collector PRIVATE ${metrics_statistics_msgs_INCLUDE_DIRS})
+target_link_libraries(system_metrics_collector ${metrics_statistics_msgs_LIBRARIES})
```

Comment:
Why is `ament_target_dependencies` not sufficient here?

Reply:
Removed `target_include_directories()` and `target_link_directories()`.